### DISCUSSION TITLE
[UI/UX] Empty State Improvements

### DISCUSSION THREAD

**Discussion by dstrukt** [btcpayserver/btcpayserver#2360] [General]
I've gone through an audited the initial empty states UI and UX, and there are a few relatively simple things I think we can do to improve this experience. There are also a few inconsistencies I've notices as well.

So for almost all of our major tabs, you're meet with this for an empty state:

## [Store] - Basic 
<img width="1842" alt="Screen Shot 2021-03-14 at 10 46 43 AM" src="https://user-images.githubusercontent.com/6250771/111078437-998afd00-84b2-11eb-8995-84e48052032d.png">

For all screens, I'm proposing that we enable a slightly more cleaned up "empty/default state" view.

<img width="1045" alt="Screen Shot 2021-03-14 at 2 10 01 PM" src="https://user-images.githubusercontent.com/6250771/111084326-044a3180-84cf-11eb-9627-aea20c9feb1c.png">

I would like to carry out that visual pattern, and the above section is pretty straightforward ...but the following have a slightly more involved interaction pattern, that feels a bit frustrating.

You need a store to utilize a Wallet, Payment Request, Invoice, App .. pretty much everything.

I've provided an example of this (which is pretty consistent across all of these).

## [Apps] - More Complex

Before
<img width="1839" alt="Screen Shot 2021-03-14 at 11 03 32 AM" src="https://user-images.githubusercontent.com/6250771/111078910-f7204900-84b4-11eb-8766-118aa8dce0e8.png">
<img width="1830" alt="Screen Shot 2021-03-14 at 11 20 48 AM" src="https://user-images.githubusercontent.com/6250771/111079469-60a15700-84b7-11eb-80c7-6dc8dd7bf759.png">

This is pretty much the pattern for the other sections.

While this will probably only happen to a user once, and hopefully they'll create a store after being prompted, this UX pattern could be improved. The introduction of our dashboard will help with this as well, but in the meantime...

Proposal:

We implement a check if a store exists, if not, preemptively display a screen that provides a "Create Store" CTA and some explanation, eliminating the additional clicks / alert modals, etc... to determine this.

After
###### (with store)
<img width="874" alt="Screen Shot 2021-03-14 at 1 31 08 PM" src="https://user-images.githubusercontent.com/6250771/111083219-a109d080-84c9-11eb-9abd-aa8d3b89ccdf.png">
###### (without store)
<img width="884" alt="Screen Shot 2021-03-14 at 1 31 13 PM" src="https://user-images.githubusercontent.com/6250771/111083220-a23afd80-84c9-11eb-922a-6a0a2f3cf505.png">

## [Invoices] - More Complex
We would treat this the same way.

Before
<img width="1835" alt="Screen Shot 2021-03-14 at 11 09 45 AM" src="https://user-images.githubusercontent.com/6250771/111083243-c3035300-84c9-11eb-9993-72b320f91aec.png">

After
<img width="811" alt="Screen Shot 2021-03-14 at 2 06 44 PM" src="https://user-images.githubusercontent.com/6250771/111084225-90a82480-84ce-11eb-9ec0-06fe221d5a44.png">
<img width="812" alt="Screen Shot 2021-03-14 at 2 06 48 PM" src="https://user-images.githubusercontent.com/6250771/111084227-9140bb00-84ce-11eb-9374-32a4fe424463.png">

## [Payment Requests] - More Complex

Before
<img width="1836" alt="Screen Shot 2021-03-14 at 1 47 57 PM" src="https://user-images.githubusercontent.com/6250771/111083674-eaf3b600-84cb-11eb-8da9-297404550130.png">

After
<img width="1164" alt="Screen Shot 2021-03-14 at 1 49 06 PM" src="https://user-images.githubusercontent.com/6250771/111084197-62c2e000-84ce-11eb-92e2-920dd6ae29de.png">
<img width="784" alt="Screen Shot 2021-03-14 at 2 05 26 PM" src="https://user-images.githubusercontent.com/6250771/111084198-63f40d00-84ce-11eb-8a9a-90e6b2ab69f1.png">

## [Wallets] - Slightly More Complex

Before
<img width="1274" alt="Screen Shot 2021-03-14 at 10 33 22 AM" src="https://user-images.githubusercontent.com/6250771/111079105-b117b500-84b5-11eb-9f08-7750a43873ac.png">

After 
<img width="852" alt="Screen Shot 2021-03-14 at 2 08 14 PM" src="https://user-images.githubusercontent.com/6250771/111084265-ce0cb200-84ce-11eb-9b14-c0e3af61851c.png">
With store created - would present the store dropdowns to kick the user into the creation flow as to prevent another redirect.
<img width="858" alt="Screen Shot 2021-03-14 at 2 08 20 PM" src="https://user-images.githubusercontent.com/6250771/111084266-cfd67580-84ce-11eb-99b7-e12161bb82ba.png">

There might be a smarter way to solve some of this, but I thought I'd raise the UX issue to prompt some discussion.

## Note:

I'm still working on how to incorporate the ( ? ) element into the views as cleanly, but it should be part of every view to link to the documentation for further learning.

Thus the designs aren't 100% finalized, but I wanted to get a signal for support / feedback before proceeding.

As always, open to any and all feedback/suggestions!

**Comment by pavlenex** [comment]
I am supportive of the new centered layout for the CTA. Though the wallet and store page may be a bit too much, it would require discussion if that kind of UX is possible without refactoring.

**Reply by dennisreimann** [reply]
i think this is perfect for the empty states – great job, @dstrukt!

**Reply by pavlenex** [reply]
If it's not too much trouble for devs, indeed would be awesome to have it on the wallet page as well. 😄

**Reply by dennisreimann** [reply]
This is so good, we'll make it work 💚

**Reply by dstrukt** [reply]
I always appreciate your thoughtfulness when approaching these changes @pavlenex, I agree the Wallet page specifically might be a little bit of a stretch, and it's not perfect, but solving the connection between Store/Wallet, with full flexibility, not just a guided set-up, etc.. has been a bit of head scratcher. 

Ultimately, I'm hoping this proposal pushes us a little closer to the most viable solution, or at least encourages some new thought towards a potentially better solution moving forward. 

I'll keep thinking on it as well, but don't hesitate to propose anything!

**Comment by Eskyee** [comment]
I think is big improvement to layout, gives us a cleaner UX

**Comment by bolatovumar** [comment]
Yeah, this is definitely better than the current empty states.

**Comment by dstrukt** [comment]
Just boosting this into the queue for any new perspectives or change of heart, as I'm probably going to try clean up my mocks a bit more, explore the Store/Wallet edge case, and try to submit a PR for it in the following week or so.

**Reply by dennisreimann** [reply]
I still like the approach and I think we can even streamline it a bit to get rid of the additional click when e.g. creating an app: Let's present the options (Name, Type and Store) on the initial form, up to three would work good imho.

Same goes for the store (Name as initial option) and invoices/payment requests (Store selection as initial option).

The rest works well as you suggested imho.

**Reply by Zaxounette** [reply]
No change or heart either, good to go on my end!

Just a note so that you think of it in your mocks: The empty state with **_no stores_** could use somewhat of a visual flash to make it more obvious. Highlighting in some way the sentence under the title should suffice.
Otherwise I sense some people will click the CTA and get sent to create a new store and wonder WTF happened and why.

**Reply by pavlenex** [reply]
It's a good improvement IMO  @dstrukt def encourage you to go for it.
